sources/packages/web/src/components/generic/DesignationAndRestrictionStatusBadge.vue

```diff
@@ -0,0 +1,65 @@
+<!--
+Used when we need to display the designation status for an Institution
+-->
+<template>
+  <v-badge :color="color" text-color="white">
+    <template v-slot:badge>
+      <font-awesome-icon :icon="icon" class="mx-1 my-1" />
+      <span class="text-uppercase">{{ label }}</span>
+    </template>
+  </v-badge>
+</template>
+<script lang="ts">
+import { onMounted, ref, watch } from "vue";
+import { DesignationAndRestrictionStatus } from "@/types";
+
+export default {
+  props: {
+    status: {
+      type: String,
+      required: true,
+    },
+  },
+  setup(props: any) {
+    const icon = ref();
+    const label = ref();
+    const color = ref();
+
+    const setIcons = () => {
+      label.value = props.status;
+      switch (props.status) {
+        case DesignationAndRestrictionStatus.designated:
+          icon.value = "check";
+          color.value = "#16C92E";
```

Comment:
when the color values are common, we can define a global constant and re-use them.